src/imitation/algorithms/preference_comparisons.py

```diff
@@ -154,16 +166,26 @@ def __init__(
         # changes the reward function), so that we return the original environment
         # rewards.
         self.buffering_wrapper = wrappers.BufferingWrapper(venv)
-        self.venv = reward_wrapper.RewardVecEnvWrapper(
+        self.venv_train = reward_wrapper.RewardVecEnvWrapper(
             self.buffering_wrapper,
-            self.reward_fn,
+            reward_fn=self.reward_fn,
         )
-        self.log_callback = self.venv.make_log_callback()
+        self.venv_eval = reward_wrapper.RewardVecEnvWrapper(
+            self.buffering_wrapper,
+            reward_fn=self.reward_fn_eval,
+        )
+        self.log_callback = self.venv_train.make_log_callback()
+
+        # venv_train should be used in self.algorithm.learn()
+        self.algorithm.set_env(self.venv_train)
 
-        self.algorithm.set_env(self.venv)
+        # TODO(yawen) Among all the attributes of venv, only venv.action_space is
```

Comment:
Huh? Surely the observation spaces need to match too (policy mapping from observation to a distribution over actions)?